### PR TITLE
Clarify what we mean by overriding phases for binary applications.

### LINKED ISSUE #878: ./configure: No such file or directory
Using the following plan file

```
pkg_origin=spuder
pkg_name=ant
pkg_version=1.9.7
pkg_maintainer="Your Name <your email address>"
pkg_license=('Apache 2')
pkg_source=http://apache.osuosl.org//ant/binaries/apache-ant-1.9.7-bin.tar.gz
pkg_shasum=1d0b808fe82cce9bcc167f8dbb7c7e89c1d7f7534c0d9c64bf615ec7c3e6bfe5
pkg_deps=()
pkg_build_deps=(
  core/make
  core/gcc
)
pkg_bin_dirs=(bin)
```

Gives the following error

    /hab/pkgs/core/hab-plan-build/0.7.0/20160614232259/bin/hab-plan-build: line 1700: ./configure: No such file or directory

```
   ant: Downloading 'http://apache.osuosl.org//ant/binaries/apache-ant-1.9.7-bin.tar.gz' to 'apache-ant-1.9.7-bin.tar.gz'
--2016-06-16 04:19:15--  http://apache.osuosl.org//ant/binaries/apache-ant-1.9.7-bin.tar.gz
Resolving apache.osuosl.org (apache.osuosl.org)... 64.50.236.52, 64.50.233.100
Connecting to apache.osuosl.org (apache.osuosl.org)|64.50.236.52|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5601575 (5.3M) [application/x-gzip]
Saving to: 'apache-ant-1.9.7-bin.tar.gz'

apache-ant-1.9.7-bi 100%[=====================>]   5.34M   509KB/s   in 9.7s

2016-06-16 04:19:25 (566 KB/s) - 'apache-ant-1.9.7-bin.tar.gz' saved [5601575/5601575]

   ant: Downloaded 'apache-ant-1.9.7-bin.tar.gz'
   ant: Verifying apache-ant-1.9.7-bin.tar.gz
   ant: Checksum verified for apache-ant-1.9.7-bin.tar.gz
   ant: Clean the cache
   ant: Unpacking apache-ant-1.9.7-bin.tar.gz
   ant: Setting build environment
mkdir: created directory '/hab/cache/src/ant-1.9.7'
   ant: Setting PREFIX=/hab/pkgs/root/ant/1.9.7/20160616041841
   ant: Setting LD_RUN_PATH=
   ant: Setting CFLAGS=-I/hab/pkgs/core/make/4.1/20160612080650/include -I/hab/pkgs/core/gcc/5.2.0/20160612064854/include
   ant: Setting LDFLAGS=-L/hab/pkgs/core/gcc/5.2.0/20160612064854/lib
   ant: Preparing to build
   ant: Building
/hab/pkgs/core/hab-plan-build/0.7.0/20160614232259/bin/hab-plan-build: line 1700: ./configure: No such file or directory
   ant: Build time: 0m44s
   ant: Exiting on error
Script done, file is /src/results/logs/..2016-06-16-041841.log
```

Same error happens when trying to use the exact redis plan. 
https://github.com/habitat-sh/habitat/blob/master/plans/redis/plan.sh

However the cpio plan works (which is nearly identical)
https://github.com/habitat-sh/habitat/blob/master/plans/cpio/plan.sh

### PR DESCRIPTION
**Pull request by juliandunn**
Closes #878.

### CODE REVIEW

**Comment by thesentinels** [comment]
By analyzing the blame information on this pull request, we identified @adamhjk to be a potential reviewer

File: www/source/docs/create-packages-binary-only.html.md [hunk 1]
```diff
@@ -10,22 +10,14 @@ You can write plans to package up these binary artifacts with minimal special ha
 
 ## Override The Phases You Don't Need
 
-A Habitat package build proceeds in phases: download, verification, unpacking (where you would also patch source code, if you had it), build, and finally installation. You can override the behavior of any of these phases by redefining the corresponding `do_` function. The following is an extreme example of overriding all the phases except install:
+A Habitat package build proceeds in phases: download, verification, unpacking (where you would also patch source code, if you had it), build, and finally installation. Each of these phases has default behavior within the build system.
```
Comment (smith) [inline www/source/docs/create-packages-binary-only.html.md]: We should link to http://localhost:4567/docs/reference/plan-syntax/#callbacks here.

**Comment by smith** [comment]
👍  looks good, but let's link back to the docs for callbacks when we talk about them.

**Comment by smith** [comment]
@thesentinels r+

**Comment by thesentinels** [comment]
:pushpin: Commit 4e1b1fc has been approved by `smith`

**Comment by thesentinels** [comment]
:hourglass: Testing commit 4e1b1fc with merge 738dddf...

**Comment by thesentinels** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/habitat-sh/habitat/builds/138939926)

**Pull request merged**